Lib/test/test_typing.py

```diff
@@ -447,6 +448,358 @@ def test_bad_var_substitution(self):
+            ('tuple[*Ts]',                             '[*Ts]',                                          'tuple[(*Ts,),]'),  # Should be tuple[*Ts]
+            ('Tuple[*Ts]',                             '[*Ts]',                                          'Tuple[*Ts]'),
+
+            ('C[*Ts]',                                 '[T, *Ts]',                                       'C[T, *Ts]'),
+            ('tuple[*Ts]',                             '[T, *Ts]',                                       'TypeError'),  # Should be tuple[T, *Ts]
+            ('Tuple[*Ts]',                             '[T, *Ts]',                                       'Tuple[T, *Ts]'),
+
+            ('C[*Ts]',                                 '[*Ts, T]',                                       'C[*Ts, T]'),
+            ('tuple[*Ts]',                             '[*Ts, T]',                                       'TypeError'),  # Should be tuple[*Ts, T]
+            ('Tuple[*Ts]',                             '[*Ts, T]',                                       'Tuple[*Ts, T]'),
+
+            ('C[T, *Ts]',                              '[int]',                                          'C[int]'),
+            ('tuple[T, *Ts]',                          '[int]',                                          'TypeError'),  # Should be tuple[int]
+            ('Tuple[T, *Ts]',                          '[int]',                                          'Tuple[int]'),
+
+            ('C[T, *Ts]',                              '[int, str]',                                     'C[int, str]'),
+            ('tuple[T, *Ts]',                          '[int, str]',                                     'tuple[int, (str,)]'),  # Should be tuple[int, str]
+            ('Tuple[T, *Ts]',                          '[int, str]',                                     'Tuple[int, str]'),
+
+            ('C[T, *Ts]',                              '[int, str, bool]',                               'C[int, str, bool]'),
+            ('tuple[T, *Ts]',                          '[int, str, bool]',                               'TypeError'),  # Should be tuple[int, str, bool]
+            ('Tuple[T, *Ts]',                          '[int, str, bool]',                               'Tuple[int, str, bool]'),
+
+            # Should raise TypeError: *tuple[int, ...] may only be used as an arg if the alias is generic[*Ts].
+            ('C[T, *Ts]',                              '[*tuple_type[int, ...]]',                        'C[*tuple_type[int, ...]]'),
+            ('tuple[T, *Ts]',                          '[*tuple_type[int, ...]]',                        'TypeError'),
+            # Should raise TypeError: *tuple[int, ...] may only be used as an arg if the alias is generic[*Ts].
+            ('Tuple[T, *Ts]',                          '[*tuple_type[int, ...]]',                        'Tuple[*tuple_type[int, ...]]'),
```

Comment:
`*tuple[int, ...]` is valid as the argument here. 

> *tuple[int, ...] may only be used as an arg if the alias is generic[*Ts].

Not true. `C[T, *Ts][*tuple[int, ...]]` is analogous to `class Foo(Generic[T, *Ts]): ...; Foo[*tuple[int, ...]]`.

Reply:
> Not true. C[T, *Ts][*tuple[int, ...]] is analogous to class Foo(Generic[T, *Ts]): ...; Foo[*tuple[int, ...]].

Yeah, sorry, should have made it clearer this was based on the tentative spec proposed in https://github.com/python/cpython/issues/91162.

I'll go through and update these comments now based on the revised tentative spec.